src/anniversary/sf33rd/Source/Game/appear.c

```diff
@@ -0,0 +1,1052 @@
+                } else {
+                    cal_all_speed_data(&wk->wu, app_counter[wk->wu.id], bg_w.bgw[1].pos_x_work - 0x58, 0, 2, 0);
+                }
+            }
+
+            wk->wu.next_z = plw[id_w].wu.my_priority;
+            break;
+
+        case 3:
+            char_move(&wk->wu);
+            app_counter[wk->wu.id]--;
+
+            if (app_counter[wk->wu.id] <= 0) {
+                wk->wu.routine_no[3]++;
+                set_char_move_init2(&wk->wu, 9, 0x11, 0x0A, 0);
+                wk->wu.next_z = wk->wu.my_priority;
+            } else {
+                add_x_sub((WORK_Other*)wk);
+            }
+
+            break;
+
+        case 4:
+            char_move(&wk->wu);
+
+            if (wk->wu.cg_type == 0xFF) {
+                wk->wu.routine_no[2] = 1;
+                wk->wu.routine_no[3] = 0;
+                Appear_end++;
+            }
+    }
+}
+
+const u8 animal_decide_tbl[] = {0, 1, 2, 3, 4, 5, 0, 2, 0, 1, 2, 3, 4, 5, 0, 0};
+
+#if defined(TARGET_PS2)
+INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/appear", Appear_37000);
+#else
+void Appear_37000(PLW *wk) {
+    not_implemented(__func__);
+}
+#endif
+
+void Appear_38000(PLW* wk) {
+    switch (wk->wu.routine_no[3]) {
+    case 0:
+        wk->wu.routine_no[3] += 1;
+        bg_app_stop = 1;
+        wk->wu.disp_flag = 1;
+
+        if (wk->wu.id) {
+            set_char_move_init(&wk->wu, 9, 0x14);
+            return;
+        }
+
+        set_char_move_init(&wk->wu, 9, 0x13);
+        break;
+
+    case 1:
+        char_move(&wk->wu);
+
+        if (wk->wu.cg_type == 0xFF) {
+            wk->wu.routine_no[2] = 1;
+            wk->wu.routine_no[3] = 0;
+            Appear_end += 1;
+        }
+    }
+}
+
+#if defined(TARGET_PS2)
+INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/appear", Appear_39000);
+#else
+void Appear_39000(PLW *wk) {
+    not_implemented(__func__);
+}
+#endif
+
+void Appear_41000(PLW* wk) {
+    switch (wk->wu.routine_no[3]) {
+    case 0:
+        wk->wu.routine_no[3]++;
+        wk->wu.disp_flag = 1;
+        bg_app_stop = 1;
+        set_char_move_init(&wk->wu, 0, 0);
+        app_counter[wk->wu.id] = 0x78;
+        effect_M7_init(wk);
+        break;
+
+    case 1:
+        char_move(&wk->wu);
+        app_counter[wk->wu.id]--;
+
+        if (app_counter[wk->wu.id] < 0) {
+            wk->wu.routine_no[2] = 1;
+            wk->wu.routine_no[3] = 0;
+            Appear_end++;
+        }
+    }
+}
+
+void gouki_appear(PLW* wk) {
+    if (!wk->wu.cmwk[0]) {
+        char_move(&wk->wu);
+        return;
+    }
+
+    switch (wk->wu.routine_no[6]) {
+    case 0:
+        wk->wu.routine_no[6] += 1;
+        set_char_move_init(&wk->wu, 1, 0x3C);
+        char_move_z(&wk->wu);
+        wk->wu.xyz[1].disp.pos = -6;
+        break;
+
+    case 1:
+        char_move(&wk->wu);
+
+        if (wk->wu.cg_type == 0xFF) {
+            wk->wu.routine_no[6] += 1;
+        }
+        /* fallthrough */
```

Comment:
Should this be a fallthrough? The last case doesn't do anything, so this looks more like a break to me
```suggestion

        break;
```

Reply:
/* fallthrough */ removed